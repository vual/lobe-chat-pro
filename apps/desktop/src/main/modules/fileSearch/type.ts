import { FileResult, SearchOptions } from '@/types/fileSearch';

/**
 * File Search Service Implementation Abstract Class
 * Defines the interface that different platform file search implementations need to implement
 */
export abstract class FileSearchImpl {
  /**
   * Perform file search
   * @param options Search options
   * @returns Promise of search result list
   */
  abstract search(options: SearchOptions): Promise<FileResult[]>;

  /**
   * Check search service status
   * @returns Promise indicating if service is available
   */
  abstract checkSearchServiceStatus(): Promise<boolean>;

  /**
   * Update search index
   * @param path Optional specified path
   * @returns Promise indicating operation success
   */
  abstract updateSearchIndex(path?: string): Promise<boolean>;
}
