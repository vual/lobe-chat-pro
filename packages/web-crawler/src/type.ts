export interface CrawlSuccessResult {
  content?: string;
  contentType: 'text' | 'json';
  description?: string;
  length?: number;
  siteName?: string;
  title?: string;
  url: string;
}

export interface CrawlErrorResult {
  content: string;
  errorMessage: string;
  errorType: string;
  url: string;
}

export interface FilterOptions {
  // 是否启用Readability
  enableReadability?: boolean;

  pureText?: boolean;
}

type CrawlImplType = 'naive' | 'jina' | 'browserless' | 'search1api';

type CrawlImplParams<T> = T & {
  filterOptions: FilterOptions;
};

export type CrawlImpl<Params = object> = (
  url: string,
  params: CrawlImplParams<Params>,
) => Promise<CrawlSuccessResult | undefined>;

export interface CrawlUrlRule {
  // 内容过滤配置（可选）
  filterOptions?: FilterOptions;
  impls?: CrawlImplType[];
  // URL匹配模式，仅支持正则表达式
  urlPattern: string;
  // URL转换模板（可选），如果提供则进行URL转换
  urlTransform?: string;
}
