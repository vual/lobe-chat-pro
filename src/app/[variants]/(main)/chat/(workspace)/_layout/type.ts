import { ReactNode } from 'react';

export interface LayoutProps {
  children: ReactNode;
  conversation: ReactNode;
  portal: ReactNode;
  topic: ReactNode;
}
