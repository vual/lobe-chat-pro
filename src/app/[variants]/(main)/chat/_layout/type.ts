import { ReactNode } from 'react';

export interface LayoutProps {
  children: ReactNode;
  session: ReactNode;
}
