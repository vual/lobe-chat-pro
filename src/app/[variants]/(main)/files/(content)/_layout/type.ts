import { ReactNode } from 'react';

export interface LayoutProps {
  children: ReactNode;
  menu: ReactNode;
  modal: ReactNode;
}
