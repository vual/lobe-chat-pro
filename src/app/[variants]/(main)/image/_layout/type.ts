import { ReactNode } from 'react';

export interface LayoutProps {
  children: ReactNode;
  menu: ReactNode;
  topic: ReactNode;
}
