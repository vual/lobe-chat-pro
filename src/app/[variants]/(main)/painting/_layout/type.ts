import { ReactNode } from 'react';

export interface LayoutProps {
  children: ReactNode;
  palette: ReactNode;
}
