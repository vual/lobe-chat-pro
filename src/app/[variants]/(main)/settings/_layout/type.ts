import { ReactNode } from 'react';

export interface LayoutProps {
  category: ReactNode;
  children: ReactNode;
}
