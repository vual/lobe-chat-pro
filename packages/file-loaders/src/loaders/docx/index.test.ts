import path from 'node:path';
import { beforeEach, describe, expect, it } from 'vitest';

import type { FileLoaderInterface } from '../../types';
import { DocxLoader } from './index';

// 确保你已经在 fixtures 目录下放置了 test.docx 文件
const fixturePath = (filename: string) => path.join(__dirname, `./fixtures/${filename}`);

let loader: FileLoaderInterface;

const testFile = fixturePath('test.docx');
const nonExistentFile = fixturePath('nonexistent.docx');

beforeEach(() => {
  loader = new DocxLoader();
});

describe('DocxLoader', () => {
  it('should load pages correctly from a DOCX file', async () => {
    const pages = await loader.loadPages(testFile);
    // DOCX 通常加载为单个页面
    expect(pages).toHaveLength(1);
    expect(pages).toMatchSnapshot();
  });

  it('should aggregate content correctly', async () => {
    const pages = await loader.loadPages(testFile);
    const content = await loader.aggregateContent(pages);
    // 对于单页文档，聚合内容应与页面内容相同
    expect(content).toEqual(pages[0].pageContent);
    expect(content).toMatchSnapshot('aggregated_content');
  });

  it('should handle file read errors in loadPages', async () => {
    const pages = await loader.loadPages(nonExistentFile);
    expect(pages).toHaveLength(1); // 即使失败也返回一个包含错误信息的页面
    expect(pages[0].pageContent).toBe('');
    expect(pages[0].metadata.error).toContain('Failed to load DOCX file');
  });
});
