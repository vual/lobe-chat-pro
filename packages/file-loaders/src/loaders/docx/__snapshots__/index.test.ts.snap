// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DocxLoader > should aggregate content correctly > aggregated_content 1`] = `
"简单报告

副标题

轻点或点按此占位符文本并开始键入即可开始。你可以在 Mac、iPad、iPhone 或 iCloud.com 上查看和编辑此文稿。

轻松编辑文本、更改字体以及添加精美的图形。使用段落样式来使整篇文稿保持一致的风格。例如，此段落使用“正文”样式。你可以在“格式”控制的“文本”标签页中更改样式。

若要添加照片、图像画廊、音频片段、视频、图表或任意 700 多种可自定义形状，请在工具栏中轻点或点按其中一个插入按钮，或者将对象拖放到页面中。你可以分层放置对象、调整其大小以及将其放在页面中的任意位置。若要更改对象随文本移动的方式，请选择对象并随后轻点或点按“格式”控制中的“排列”标签页。

小标题

Pages 文稿可用于文字处理和页面布局。此“简单报告”模板为文字处理而设置，如此一来，文本便会随着你的键入而从某一页流向下一页，到达页面末尾时会自动创建新的页面。

在页面布局文稿中，你可以手动重新排列页面并随意调整页面中的文本框、图像和其他对象的位置。若要创建页面布局文稿，请在模板选取器中选取一种页面布局模板。你也可以在 Mac、iPad 或 iPhone 上将此文稿改为页面布局，方法是在“文稿”控制中关闭“文稿正文”。

“这是一个引用（报告中的关键短语）的例子。轻点或点按此文本添加你自己的内容。”









这是第二页的内容

"
`;

exports[`DocxLoader > should load pages correctly from a DOCX file 1`] = `
[
  {
    "charCount": 587,
    "lineCount": 29,
    "metadata": {
      "pageNumber": 1,
    },
    "pageContent": "简单报告

副标题

轻点或点按此占位符文本并开始键入即可开始。你可以在 Mac、iPad、iPhone 或 iCloud.com 上查看和编辑此文稿。

轻松编辑文本、更改字体以及添加精美的图形。使用段落样式来使整篇文稿保持一致的风格。例如，此段落使用“正文”样式。你可以在“格式”控制的“文本”标签页中更改样式。

若要添加照片、图像画廊、音频片段、视频、图表或任意 700 多种可自定义形状，请在工具栏中轻点或点按其中一个插入按钮，或者将对象拖放到页面中。你可以分层放置对象、调整其大小以及将其放在页面中的任意位置。若要更改对象随文本移动的方式，请选择对象并随后轻点或点按“格式”控制中的“排列”标签页。

小标题

Pages 文稿可用于文字处理和页面布局。此“简单报告”模板为文字处理而设置，如此一来，文本便会随着你的键入而从某一页流向下一页，到达页面末尾时会自动创建新的页面。

在页面布局文稿中，你可以手动重新排列页面并随意调整页面中的文本框、图像和其他对象的位置。若要创建页面布局文稿，请在模板选取器中选取一种页面布局模板。你也可以在 Mac、iPad 或 iPhone 上将此文稿改为页面布局，方法是在“文稿”控制中关闭“文稿正文”。

“这是一个引用（报告中的关键短语）的例子。轻点或点按此文本添加你自己的内容。”









这是第二页的内容

",
  },
]
`;
